File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {

        int l = 0;
        int r = nums.size() - 1;
        int ans = -1;
        while(l<=r){
            int mid = l + (r-l) / 2;
            if(nums[mid] == target){
                return mid;
            }
            // for left
            else if(nums[l] <= nums[mid]){
                if(target >= nums[l] && target < nums[mid]){
                    r = mid - 1;
                }
                else{
                    l = mid + 1;
                }
            }
            // for right
            else{
                if(target <= nums[r] && target > nums[mid]){
                    l = mid + 1;
                } 
                else{
                    r = mid -1;
                }
            }
        }
    return ans;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        curr = root
        q = deque()
        q.append(curr)
        ans = []
        while q:
            qlen = len(q)
            val = None
            for i in range(qlen):
                node = q.popleft()
                if node:
                    val = node.val
                    q.append(node.left)
                    q.append(node.right)

            if val:
                ans.append(val)  
        return ans
        